Windows compilation error "config.h: No such file or directory"

`Can you help me with this problem?`

D:\download\n2n-2.8>md build

D:\download\n2n-2.8>cd build

D:\download\n2n-2.8\build>cmake ..
-- Building for: Visual Studio 15 2017
-- Selecting Windows SDK version 10.0.17763.0 to target Windows 10.0.18363.
-- The C compiler identification is MSVC 19.16.27035.0
-- The CXX compiler identification is MSVC 19.16.27035.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/VC/Tools/MSVC/14.16.27023/bin/Hostx86/x86/cl.exe -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Configuring done
-- Generating done
-- Build files have been written to: D:/download/n2n-2.8/build

D:\download\n2n-2.8\build>"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin\amd64\MSBuild.exe" edge.vcxproj /t:Build /p:Configuration=Release
Microsoft (R) Build Engine verze 15.9.21+g9802d43bc3 pro .NET Framework
Copyright (C) Microsoft Corporation. Všechna práva vyhrazena.

Vytváření sestavení bylo zahájeno 16.8.2020 9:35:00.
Projekt D:\download\n2n-2.8\build\edge.vcxproj v uzlu 1 (počet cílů: Build).
Probíhá sestavení projektu D:\download\n2n-2.8\build\edge.vcxproj (1) - D:\download\n2n-2.8\build\ZERO_CHECK.vcxproj (2) v uzlu 1 (výchozí cíle).
PrepareForBuild:
  Probíhá vytváření adresáře Win32\Release\ZERO_CHECK\.
  Probíhá vytváření adresáře D:\download\n2n-2.8\build\Release\.
  Probíhá vytváření adresáře Win32\Release\ZERO_CHECK\ZERO_CHECK.tlog\.
InitializeBuildStatus:
  Probíhá vytváření souboru Win32\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild v důsledku zadání AlwaysCreate.
CustomBuild:
  Checking Build System
FinalizeBuildStatus:
  Probíhá odstraňování souboru Win32\Release\ZERO_CHECK\ZERO_CHECK.tlog\unsuccessfulbuild.
  Probíhá aktualizace časového razítka pro Win32\Release\ZERO_CHECK\ZERO_CHECK.tlog\ZERO_CHECK.lastbuildstate.
Sestavení projektu D:\download\n2n-2.8\build\ZERO_CHECK.vcxproj (s výchozími cíli) bylo dokončeno.

Probíhá sestavení projektu D:\download\n2n-2.8\build\edge.vcxproj (1) - D:\download\n2n-2.8\build\edge_utils_win32.vcxproj (3) v uzlu 1 (výchozí cíle).
PrepareForBuild:
  Probíhá vytváření adresáře edge_utils_win32.dir\Release\.
  Probíhá vytváření adresáře edge_utils_win32.dir\Release\edge_utils_win32.tlog\.
InitializeBuildStatus:
  Probíhá vytváření souboru edge_utils_win32.dir\Release\edge_utils_win32.tlog\unsuccessfulbuild v důsledku zadání AlwaysCreate.
CustomBuild:
  Building Custom Rule D:/download/n2n-2.8/CMakeLists.txt
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86\CL.exe /c /I"C:\Program Files (x86)\OpenSSL-Win32\include" /I"D:\download\n2n-2.8\." /I"D:\download\n2n-2.8\include" /I"D:\download\n2n-2.8\win32" /W3 /WX- /diagnostics:classic /O2 /Ob2 /Oy- /D WIN32 /D _WINDOW
  S /D NDEBUG /D CMAKE_BUILD /D "GIT_RELEASE=\"\"" /D "PACKAGE_VERSION=\"2.7.0\"" /D "PACKAGE_OSNAME=\"Windows-10.0.18363\"" /D "N2N_VERSION=\"2.7.0\"" /D "N2N_OSNAME=\"Windows-10.0.18363\"" /D N2N_HAVE_AES /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"edge_
  utils_win32.dir\Release\\" /Fd"edge_utils_win32.dir\Release\edge_utils_win32.pdb" /Gd /TC /analyze- /errorReport:queue "D:\download\n2n-2.8\src\edge_utils_win32.c"
  Optimalizující kompilátor C/C++ společnosti Microsoft (R) verze 19.16.27035 pro procesory x86
  Copyright (C) Microsoft Corporation. Všechna práva vyhrazena.

  cl /c /I"C:\Program Files (x86)\OpenSSL-Win32\include" /I"D:\download\n2n-2.8\." /I"D:\download\n2n-2.8\include" /I"D:\download\n2n-2.8\win32" /W3 /WX- /diagnostics:classic /O2 /Ob2 /Oy- /D WIN32 /D _WINDOWS /D NDEBUG /D CMAKE_BUILD /D "GIT_RELEASE=\"\"" /D "PACKAGE_VERSION=\"2.7.0\"" /D "PACKAGE_OSNAME=\"Windows
  -10.0.18363\"" /D "N2N_VERSION=\"2.7.0\"" /D "N2N_OSNAME=\"Windows-10.0.18363\"" /D N2N_HAVE_AES /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"edge_utils_win32.dir\Release\\" /Fd"edge_utils_win32.dir\Release\edge_utils_win32.pdb" /Gd /TC /analyze- /errorRe
  port:queue "D:\download\n2n-2.8\src\edge_utils_win32.c"

  edge_utils_win32.c
D:\download\n2n-2.8\include\n2n.h(42): fatal error C1083: Nejde otevřít soubor zahrnout: config.h: No such file or directory [D:\download\n2n-2.8\build\edge_utils_win32.vcxproj]
Sestavení projektu D:\download\n2n-2.8\build\edge_utils_win32.vcxproj (s výchozími cíli) bylo dokončeno: CHYBA.

Probíhá sestavení projektu D:\download\n2n-2.8\build\edge.vcxproj (1) - D:\download\n2n-2.8\build\n2n.vcxproj (4) v uzlu 1 (výchozí cíle).
Probíhá sestavení projektu D:\download\n2n-2.8\build\n2n.vcxproj (4) - D:\download\n2n-2.8\build\win32\n2n_win32.vcxproj (5) v uzlu 1 (výchozí cíle).
PrepareForBuild:
  Probíhá vytváření adresáře n2n_win32.dir\Release\.
  Probíhá vytváření adresáře n2n_win32.dir\Release\n2n_win32.tlog\.
InitializeBuildStatus:
  Probíhá vytváření souboru n2n_win32.dir\Release\n2n_win32.tlog\unsuccessfulbuild v důsledku zadání AlwaysCreate.
CustomBuild:
  Building Custom Rule D:/download/n2n-2.8/win32/CMakeLists.txt
ClCompile:
  C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.16.27023\bin\HostX86\x86\CL.exe /c /I"C:\Program Files (x86)\OpenSSL-Win32\include" /I"D:\download\n2n-2.8\." /I"D:\download\n2n-2.8\include" /I"D:\download\n2n-2.8\win32" /W3 /WX- /diagnostics:classic /O2 /Ob2 /Oy- /D WIN32 /D _WINDOW
  S /D NDEBUG /D CMAKE_BUILD /D "GIT_RELEASE=\"\"" /D "PACKAGE_VERSION=\"2.7.0\"" /D "PACKAGE_OSNAME=\"Windows-10.0.18363\"" /D "N2N_VERSION=\"2.7.0\"" /D "N2N_OSNAME=\"Windows-10.0.18363\"" /D N2N_HAVE_AES /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"n2n_w
  in32.dir\Release\\" /Fd"n2n_win32.dir\Release\n2n_win32.pdb" /Gd /TC /analyze- /errorReport:queue "D:\download\n2n-2.8\win32\getopt1.c" "D:\download\n2n-2.8\win32\getopt.c" "D:\download\n2n-2.8\win32\wintap.c"
  Optimalizující kompilátor C/C++ společnosti Microsoft (R) verze 19.16.27035 pro procesory x86
  Copyright (C) Microsoft Corporation. Všechna práva vyhrazena.

  getopt1.c
  cl /c /I"C:\Program Files (x86)\OpenSSL-Win32\include" /I"D:\download\n2n-2.8\." /I"D:\download\n2n-2.8\include" /I"D:\download\n2n-2.8\win32" /W3 /WX- /diagnostics:classic /O2 /Ob2 /Oy- /D WIN32 /D _WINDOWS /D NDEBUG /D CMAKE_BUILD /D "GIT_RELEASE=\"\"" /D "PACKAGE_VERSION=\"2.7.0\"" /D "PACKAGE_OSNAME=\"Windows
  -10.0.18363\"" /D "N2N_VERSION=\"2.7.0\"" /D "N2N_OSNAME=\"Windows-10.0.18363\"" /D N2N_HAVE_AES /D "CMAKE_INTDIR=\"Release\"" /D _MBCS /Gm- /MD /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /Fo"n2n_win32.dir\Release\\" /Fd"n2n_win32.dir\Release\n2n_win32.pdb" /Gd /TC /analyze- /errorReport:queue "D:\downlo
  ad\n2n-2.8\win32\getopt1.c" "D:\download\n2n-2.8\win32\getopt.c" "D:\download\n2n-2.8\win32\wintap.c"

  getopt.c
  wintap.c
D:\download\n2n-2.8\include\n2n.h(42): fatal error C1083: Nejde otevřít soubor zahrnout: config.h: No such file or directory [D:\download\n2n-2.8\build\win32\n2n_win32.vcxproj]
  Generování kódu...
Sestavení projektu D:\download\n2n-2.8\build\win32\n2n_win32.vcxproj (s výchozími cíli) bylo dokončeno: CHYBA.

Sestavení projektu D:\download\n2n-2.8\build\n2n.vcxproj (s výchozími cíli) bylo dokončeno: CHYBA.

Sestavení projektu D:\download\n2n-2.8\build\edge.vcxproj (počet cílů: Build) bylo dokončeno: CHYBA.


Sestavení SE NEZDAŘILO.

"D:\download\n2n-2.8\build\edge.vcxproj" (Build cíl) (1) ->
"D:\download\n2n-2.8\build\edge_utils_win32.vcxproj" (výchozí cíl) (3) ->
(cíl ClCompile) ->
  D:\download\n2n-2.8\include\n2n.h(42): fatal error C1083: Nejde otevřít soubor zahrnout: config.h: No such file or directory [D:\download\n2n-2.8\build\edge_utils_win32.vcxproj]


"D:\download\n2n-2.8\build\edge.vcxproj" (Build cíl) (1) ->
"D:\download\n2n-2.8\build\n2n.vcxproj" (výchozí cíl) (4) ->
"D:\download\n2n-2.8\build\win32\n2n_win32.vcxproj" (výchozí cíl) (5) ->
  D:\download\n2n-2.8\include\n2n.h(42): fatal error C1083: Nejde otevřít soubor zahrnout: config.h: No such file or directory [D:\download\n2n-2.8\build\win32\n2n_win32.vcxproj]

    0 upozornění
    Počet chyb: 2

Uplynulý čas 00:00:02.27

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,7 +4,7 @@ include(CheckFunctionExists)
 SET(CMAKE_VERBOSE_MAKEFILE ON)
 
 # N2n release information
-set(N2N_VERSION "2.7.0")
+set(N2N_VERSION "2.8.0")
 set(N2N_OSNAME ${CMAKE_SYSTEM})
 execute_process(
         COMMAND git --version

--- a/include/n2n.h
+++ b/include/n2n.h
@@ -38,7 +38,7 @@
 #ifdef WIN32
 #include "win32/n2n_win32.h"
 
-#ifdef _MSC_VER
+#ifndef CMAKE_BUILD
 #include "config.h" /* Visual C++ */
 #else
 #include "win32/winconfig.h"

